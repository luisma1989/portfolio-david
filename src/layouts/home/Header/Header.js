import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import WOW from 'wowjs'
import { Parallax } from 'react-scroll-parallax'
import { H4 } from 'components/Fonts'
import {
  Container,
  ParallaGradient,
  Wrapper,
  Title,
} from './styles'

const Header = ({ header, getHeaderData, intl }) => {
  useEffect(async () => {
    await getHeaderData()
    new WOW.WOW().init()
  }, [])
  const name = header.name || intl.messages.header.name
  const apellidos = header.apellidos || intl.messages.header.apellidos

  return (
    <Wrapper>
      <Container className='header'>
        <ParallaGradient>
          {/* <ChangeLanguage switchLanguage={lang => this.props.dispatch(switchLanguage(lang))} intl={this.props.intl} />*/}
          <Parallax offsetYMax={40} offsetYMin={0} slowerScrollRate tag='figure'>
            <div className='homeContainer'>
              <div className='home_text'>
                <Title variant='white' className='title'>
                  {name} {apellidos}
                </Title>
              </div>
            </div>
          </Parallax>
        </ParallaGradient>
      </Container>
      <H4>Chef ejecutivo</H4>
    </Wrapper>
  )
}

Header.propTypes = {
  header: PropTypes.object.isRequired,
}

export default Header
