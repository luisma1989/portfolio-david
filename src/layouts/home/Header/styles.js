import styled from 'styled-components'
import bg from './assets/bg.jpg'
import { black, white } from 'utils/constants/colors'
import { H1 } from 'components/Fonts'

export const Wrapper = styled.section`
  overflow: hidden;
  max-width: 1500px;
  margin: 0 auto;
  height: 100vh;
  background-color: ${black};
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;

  h4 {
    position: absolute;
    bottom: 20px;
    right: 40px;
    color: ${white};
  }
`;

export const Container = styled.div`
  background: url(${bg});
  background-size: cover;
  background-repeat: no-repeat;
  width: 60rem;
  height: 60rem;
  border-radius: 50%;
  background-position: center;

  @media (max-width: 860px) {
    width: 50rem;
    height: 50rem;
  }

  @media (max-width: 640px) {
    width: 35rem;
    height: 35rem;
  }

  .parallax-inner {
    transform: initial!important;
  }
`;

export const ParallaGradient = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Title = styled(H1)`
  font-family: 'Righteous', sans-serif;
  text-transform: uppercase;
  font-size: 85px;
  white-space: nowrap;

  @media (max-width: 1220px) {
    font-size: 75px;
  }

  @media (max-width: 1080px) {
    font-size: 80px;
    white-space: initial;
  }

  @media (max-width: 860px) {
    font-size: 62px;
  }
`;
