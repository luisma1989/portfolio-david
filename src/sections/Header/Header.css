* {
  padding: 0;
  margin: 0; }

body {
  font-family: 'Raleway', sans-serif;
  overflow-y: scroll;
  overflow-x: hidden; }

.header {
  background: url(./assets/bg.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  transition: transform 5000ms; }
  .header:hover {
    transform: scale(1.1); }
  .header .parallax-gradient {
    background: radial-gradient(ellipse farthest-corner at center center, rgba(0, 0, 0, 0.5) 20%, rgba(0, 0, 0, 0.85) 100%) repeat scroll 0% 0%; }

.home_text {
  text-align: left; }

.home_text h1 {
  font-family: 'Josefin Slab', serif;
  font-size: 60px;
  color: darkgrey;
  text-transform: uppercase;
  text-align: center; }
  .home_text h1 span {
    color: #fff;
    font-size: 100px; }

.home_text p {
  color: #fff;
  font-size: 17px;
  line-height: 28px;
  margin-top: 0px;
  font-weight: 300; }

@media only screen and (max-width: 640px) {
  .home_text h1 {
    font-size: 45px; }
    .home_text h1 span {
      font-size: 75px; } }
