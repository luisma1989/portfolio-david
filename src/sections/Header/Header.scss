* {
  padding: 0;
  margin: 0;
}

body {
  font-family: 'Raleway', sans-serif;
  overflow-y: scroll;
  overflow-x: hidden;
}

.header {
  margin: 50px;
  overflow: hidden;
  background: url(./assets/bg.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  transition: transform 5000ms;

  &:hover {
    transform: scale(1.1);
  }

  .parallax-gradient {
    background: radial-gradient(ellipse farthest-corner at center center, rgba(0, 0, 0, 0.5) 20%, rgba(0, 0, 0, 0.85) 100%) repeat scroll 0% 0%;
  }
}


.home_text {
  text-align: left;
}

.home_text {
  h1 {
    font-family: 'Josefin Slab', serif;
    font-size: 60px;
    color: darkgrey;
    text-transform: uppercase;
    text-align: center;

    span {
      color: #fff;
      font-size: 100px;
    }
  }
}

.home_text p {
  color: #fff;
  font-size: 17px;
  line-height: 28px;
  margin-top: 0px;
  font-weight: 300;
}

@media only screen and (max-width: 640px) {
  .home_text {
    h1 {
      font-size: 45px;

      span {
        font-size: 75px;
      }
    }
  }
}
