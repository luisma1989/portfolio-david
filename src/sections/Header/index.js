import React from 'react'
import PropTypes from 'prop-types'
import WOW from 'wowjs'
import { Parallax } from 'react-scroll-parallax'
import { H4 } from 'components/Fonts'
// import ChangeLanguage from '../../components/ChangeLanguage'
// import { switchLanguage } from '../../Intl/IntlActions'
import {
  Container,
  ParallaGradient,
  Wrapper,
  Title,
} from './styles'

class Header extends React.Component {
  static propTypes = {
    headerData: PropTypes.object.isRequired,
  }

  componentDidMount () {
    new WOW.WOW().init()
  }

  render () {
    const { name, apellidos } = this.props.headerData

    return (
      <Wrapper>
        <Container className='header'>
          <ParallaGradient>
            {/* <ChangeLanguage switchLanguage={lang => this.props.dispatch(switchLanguage(lang))} intl={this.props.intl} />*/}
            <Parallax offsetYMax={40} offsetYMin={0} slowerScrollRate tag='figure'>
              <div className='homeContainer'>
                <div className='home_text'>
                  <Title variant='white' className='wow rubberBand title'>
                    {name} {apellidos}
                  </Title>
                </div>
              </div>
            </Parallax>
          </ParallaGradient>
        </Container>
        <H4>Chef ejecutivo</H4>
      </Wrapper>
    )
  }
}

export default Header
