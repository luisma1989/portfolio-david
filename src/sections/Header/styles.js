import styled from 'styled-components'
import bg from './assets/bg.jpg'
import { black, white } from 'utils/constants/colors'
import { H1 } from 'components/Fonts'

export const Wrapper = styled.section`
  overflow: hidden;
  max-width: 1500px;
  margin: 0 auto;
  height: 100vh;
  background-color: ${black};
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;

  h4 {
    position: absolute;
    bottom: 20px;
    right: 40px;
    color: ${white};
  }
`;

export const Container = styled.div`
  background: url(${bg});
  background-size: cover;
  background-repeat: no-repeat;
  width: 60rem;
  height: 60rem;
  border-radius: 50%;
  background-position: center;
`;

export const ParallaGradient = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Title = styled(H1)`
  font-family: 'Righteous', sans-serif;
  text-transform: uppercase;
  font-size: 85px;
  white-space: nowrap;
`;
