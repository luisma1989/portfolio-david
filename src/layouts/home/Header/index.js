import { connect } from 'react-redux'
import Header from './Header'
import { getHeaderData } from 'reducers/header'

// Retrieve data from store as props
const mapStateToProps = ({ intl, header }) => {
  return {
    intl,
    header,
  }
}

const mapDispatchToProps = {
  getHeaderData,
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)
