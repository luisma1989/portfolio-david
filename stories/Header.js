import React from 'react';
import { storiesOf } from '@storybook/react';
import Header from 'sections/Header';

const headerData = {
  headerData:{
    name: 'name',
    apellidos: 'apellidos',
  }
}

storiesOf('Header', module)
  .add('Component', () =>
    <Header headerData={headerData} />
  )
