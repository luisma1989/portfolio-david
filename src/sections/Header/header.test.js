// eslint-disable-next-line react/react-in-jsx-scope
import React from 'react';
import Header from './index.js'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const headerData = {
  headerData: {
    name: 'name',
    apellidos: 'apellidos',
  }
}

describe('first test', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Header headerData={headerData} />)
  });
  it('Shoul render the component', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('Should render the profession', () => {
    const H4 = wrapper.find('Fonts__H4')

    expect(H4.children().text()).toBe('Chef ejecutivo');
  });

});