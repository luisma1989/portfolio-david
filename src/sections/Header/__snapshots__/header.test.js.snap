// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first test Shoul render the component 1`] = `
<styles__Wrapper>
  <styles__Container
    className="header"
  >
    <styles__ParallaGradient>
      <Parallax
        disabled={false}
        offsetXMax={0}
        offsetXMin={0}
        offsetYMax={40}
        offsetYMin={0}
        slowerScrollRate={true}
        tag="figure"
      >
        <div
          className="homeContainer"
        >
          <div
            className="home_text"
          >
            <styles__Title
              className="wow rubberBand title"
              variant="white"
            >
               
            </styles__Title>
          </div>
        </div>
      </Parallax>
    </styles__ParallaGradient>
  </styles__Container>
  <Fonts__H4>
    Chef ejecutivo
  </Fonts__H4>
</styles__Wrapper>
`;
